File: test/fixtures/go/Advanced.go
package main

// 泛型栈
type Stack[T any] struct {
	// 元素列表
	items []T
}

// 入栈
func (s *Stack[T]) Push(item T) {
	s.items = append(s.items, item)
}

// 出栈
func (s *Stack[T]) Pop() T {
	if len(s.items) == 0 {
		var zero T
		return zero
	}
	item := s.items[len(s.items)-1]
	s.items = s.items[:len(s.items)-1]
	return item
}

// 嵌入基类
type Base struct {
	// 名称
	Name string
}

// 派生结构
type Derived struct {
	Base
	// 映射表
	Lookup map[string]int
}

File: test/fixtures/kotlin/Advanced.kt
/**
 * 数据类
 */
data class UserInfo(
    /** 用户ID */
    val id: Int,
    /** 用户名 */
    val name: String,
)

/**
 * 泛型仓库
 */
class Repo<T> {
    /** 延迟初始化 */
    lateinit var items: List<T>

    /** 保存项目 */
    fun save(item: T): Boolean {
        return true
    }
}

/** 字符串扩展 */
fun String.shout(): String {
    return this.uppercase()
}
